module-system/module-implementations/sov-blob-storage/src/capabilities.rs

```diff
@@ -0,0 +1,57 @@
+use sov_modules_api::capabilities::BlobSelector;
+use sov_modules_api::{Context, Spec};
+use sov_rollup_interface::da::BlobReaderTrait;
+use sov_state::WorkingSet;
+
+use crate::BlobStorage;
+
+impl<C: Context> BlobSelector for BlobStorage<C> {
+    type Context = C;
+
+    fn get_blobs_for_this_slot<'a, I, B>(
+        &self,
+        current_slot: u64,
+        current_blobs: I,
+        pull_from_slot: u64,
+        working_set: &mut WorkingSet<<Self::Context as Spec>::Storage>,
+    ) -> (Vec<&'a mut B>, Vec<B>)
```

Comment:
It seems a little odd that we return two separate `Vec`s here; feels like the trait matches very specifically with our initial use case - which makes me wonder if it should be a trait at all. As it is, when we're not using the deferred blob feature we'll always return `(current_blobs, vec[]!)` - which seems a little odd.

Maybe we can make a more general interface like:

```
fn get_blobs_for_this_slot<'a, I, B>(
        &self,
        current_slot: u64,
        current_blobs: I,
        working_set: &mut WorkingSet<<Self::Context as Spec>::Storage>,
    ) -> Vec<B>
```
    
  I see there's a slight mismatch between the borrowed and owned `Blob` types, but we can hide that behind an enum or a trait.

Reply:
I decided that `enum` is a way to go, quite straighforward, also it can implement public method to return mutable reference, so